server.go

```diff
@@ -446,7 +448,16 @@ func (s *server) handshake(u *User) error {
 			continue
 		}
 		s.u = u
-		return s.welcome(u)
+
+		err = s.welcome(u)
+		if err == nil && u.Pass != nil {
+			service := "mattermost"
+			if len(u.Pass) == 1 {
+				service = "slack"
+			}
+			login(u, u, u.Pass, service)
```

Comment:
I would change this to
login(u, &User{Nick: service, User: service, Real: service, Host: "service", channels: map[Channel]struct{}{}}, u.Pass, service)
The login OK message needs to come from the service, not the user